### PR TITLE
React ip opt controller fix me

### PR DESCRIPTION
**Pull request by brahmgardner**
new function `initializeApp` called in the constructor of `ctrlThread` so that it is only initialized once

### CODE REVIEW

File: lib/include/react_controller/ctrlThread.h [hunk 1]
```diff
@@ -34,6 +36,8 @@ class CtrlThread : public RobotInterface
     CtrlThread(const std::string& _name, const std::string& _limb, bool _no_robot,
                bool _is_debug = false, double dT = 0.02, double tol = 1e-6, double vMax = 45.0);
 
+    void initializeApp(bool _verbosity);
```
Comment (alecive) [inline lib/include/react_controller/ctrlThread.h]: Add documentation of the initializeapp.

File: lib/src/react_controller/controllerNLP.cpp [hunk 2]
```diff
@@ -11,8 +11,7 @@ using namespace   std;
 ControllerNLP::ControllerNLP(BaxterChain chain_, double dt_, bool ctrl_ori_) :
                                 chain(chain_), dt(dt_), ctrl_ori(ctrl_ori_)
 {
-    xr.resize(6); xr.setZero();
-    set_xr(xr);
+    xr.resize(6);
```
Comment (alecive) [inline lib/src/react_controller/controllerNLP.cpp]: why you don't do set_xr any more?

File: lib/src/react_controller/ctrlThread.cpp [hunk 3]
```diff
@@ -60,6 +69,26 @@ CtrlThread::CtrlThread(const std::string& _name, const std::string& _limb, bool
...
+    if (is_debug == false) {
+        app->Options()->SetStringValue ("linear_solver", "ma57");
+    }
+    app->Options()->SetIntegerValue("max_iter",std::numeric_limits<int>::max());
+    app->Options()->SetNumericValue("max_cpu_time", 0.95 * dT);
+    // app->Options()->SetStringValue ("nlp_scaling_method","gradient-based");
+    app->Options()->SetStringValue ("hessian_approximation","limited-memory");
+    // app->Options()->SetStringValue ("derivative_test",verbosity?"first-order":"none");
+    app->Options()->SetStringValue ("derivative_test","none");
+    app->Options()->SetIntegerValue("print_level",_verbosity & !is_debug?5:0);
```
Comment (alecive) [inline lib/src/react_controller/ctrlThread.cpp]: What codacy said is correct, please clarify calculation precedence for '+' and '?'.

File: lib/src/react_controller/ctrlThread.cpp [hunk 4]
```diff
@@ -195,38 +225,20 @@ VectorXd CtrlThread::solveIK(int &_exit_code)
     xr.block<3, 1>(0, 0) = x_n;
     xr.block<3, 1>(3, 0) = o_n;
 
-    MatrixXd vLimAdapted(DoFs, 2);
-    for (size_t r = 0; r < DoFs; ++r)
+
+    Ipopt::SmartPtr<ControllerNLP> nlp = new ControllerNLP(*chain);
+    nlp->set_ctrl_ori(false);
+    nlp->set_dt(dT);
```
Comment (alecive) [inline lib/src/react_controller/ctrlThread.cpp]: I would put the initialization of the controllerNLP after you set vlimadapted

File: lib/src/react_controller/controllerNLP.cpp (hunk 2 above)
Comment (brahmgardner) [inline lib/src/react_controller/controllerNLP.cpp]: Fixed.

**Pull request merged**